pagerdutyplugin/data_source_pagerduty_priority.go

```diff
@@ -0,0 +1,98 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/PagerDuty/go-pagerduty"
+	"github.com/PagerDuty/terraform-provider-pagerduty/util"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+)
+
+type dataSourcePriority struct{ client *pagerduty.Client }
+
+var _ datasource.DataSourceWithConfigure = (*dataSourcePriority)(nil)
+
+func (*dataSourcePriority) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "pagerduty_priority"
+}
+
+func (*dataSourcePriority) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{Computed: true},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "The name of the priority to find in the PagerDuty API",
+			},
+			"description": schema.StringAttribute{Computed: true},
+		},
+	}
+}
+
+func (d *dataSourcePriority) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	resp.Diagnostics.Append(ConfigurePagerdutyClient(&d.client, req.ProviderData)...)
+}
+
+func (d *dataSourcePriority) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	log.Println("[INFO] Reading PagerDuty priority")
+
+	var searchName types.String
+	resp.Diagnostics.Append(req.Config.GetAttribute(ctx, path.Root("name"), &searchName)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var found *pagerduty.Priority
+	err := retry.RetryContext(ctx, 2*time.Minute, func() *retry.RetryError {
+		list, err := d.client.ListPrioritiesWithContext(ctx, pagerduty.ListPrioritiesOptions{})
+		if err != nil {
+			if util.IsBadRequestError(err) {
+				return retry.NonRetryableError(err)
+			}
+			return retry.RetryableError(err)
+		}
+
+		for _, priority := range list.Priorities {
```

Comment:
`list.Priorities` only has the items from first page of results. Could you please make this to loop through all the pages to afterwards move forward with the lookup?